fastparquet/compression.py

```diff
@@ -1,15 +1,43 @@
 
 import gzip
 from .thrift_structures import parquet_thrift
+from .util import PY2
+
 # TODO: use stream/direct-to-buffer conversions instead of memcopy
 
 # TODO: enable ability to pass kwargs to compressor
 
-compressions = {'GZIP': gzip.compress,
-                'UNCOMPRESSED': lambda x: x}
+compressions = {
+    'UNCOMPRESSED': lambda x: x
+}
+decompressions = {
+    'UNCOMPRESSED': lambda x: x
+}
+
+# Gzip is present regardless
+COMPRESSION_LEVEL = 9
+if PY2:
+    def gzip_compress_v2(data, compresslevel=COMPRESSION_LEVEL):
+        from io import BytesIO
+        bio = BytesIO()
+        f = gzip.GzipFile(mode='wb',
+                          compresslevel=compresslevel,
+                          fileobj=bio)
+        f.write(data)
+        f.close()
+        return bio.getvalue()
+    def gzip_decompress_v2(data):
+        import zlib
+        return zlib.decompress(data,
+                               16+15)
```

Comment:
I agree it's a bit arcane.  It has to do with zlib 'windowBits' parameter.   From the zlib manual:

The windowBits parameter is the base two logarithm of the window size (the size of the history buffer). It should be in the range 8..15 for this version of the library. Larger values of this parameter result in better compression at the expense of memory usage. The default value is 15 if deflateInit is used instead.

The '16' is a base value that indicates that gzip decompression must take place.  The '15' is a maximum window size, as indicated above.

I can add a brief comment to the code.